lib/DB/Schema/Result/UserProblem.pm

```diff
@@ -22,74 +21,77 @@ __PACKAGE__->add_columns(
 	user_problem_id => {
 		data_type         => 'integer',
 		size              => 16,
-		is_nullable       => 0,
 		is_auto_increment => 1,
 	},
 	set_problem_id => {
-		data_type   => 'integer',
-		size        => 16,
-		is_nullable => 0,
+		data_type => 'integer',
+		size      => 16,
 	},
 	user_set_id => {
-		data_type   => 'integer',
-		size        => 16,
-		is_nullable => 0,
+		data_type => 'integer',
+		size      => 16,
 	},
 	seed => {
-		data_type   => 'integer',
-		size        => 16,
-		is_nullable => 0,
+		data_type => 'integer',
+		size      => 16,
 	},
 	status => {
 		data_type     => 'decimal',
 		size          => '10,5',
-		is_nullable   => 0,
 		default_value => '0.00000'
```

Comment:
It seems like for all of the fields with default_value, we are now also setting `retrieve_on_insert => 1`.  Should that go here too?

Reply:
good catch, I missed one! (this means we can stop worrying about `->discard_changes`